### DISCUSSION TITLE
서비스 로그의 표준을 만들어나가고 퀄리티를 유지하기 위해서는 어떻게 해야할까요?

### DISCUSSION THREAD

**Discussion by jongikp** [playinpap/awesome-data-and-analytics-governance#18] [Q&A]
안녕하세요. 궁금한게 있어서 글을 남겨봅니다!
프로덕트를 출시할 때마다 혹은 개선시킬 때마다 web 혹은 client 로그 를 설계하고 세팅합니다.

그러면서 항상 문득 드는 생각은 이러한 작업은 단순히 데이터팀이 아닌 사람들도 할 수 있는 일이 아닌가 싶습니다.
따라서 본인이 원하는 서비스 데이터를 직접 설계하고 개발 요청하여 스스로 분석을 하는 모습이 정말 아름다운 모습이라고 생각하는데요.

그렇게 하기 위해서는 전사적으로 표준화된 로그 포맷과 프로세스가 있어야할 것 같습니다.
더 나아가서는 로그 개발 자동화를 통해, 생산 속도와 일정한 컨벤션을 시스템을 통해 관리되어야한다고 생각합니다.

그랬을 때, 누구나 원하는 데이터를 설계해서 분석하는 모습을 이뤄낼 수 있다고 생각하는데,
그 것의 첫 단계가 표준화된 로그 포맷이라고 생각합니다.

혹시 여러분들은,
어떠한 방법과 관점으로 로그 포맷의 표준을 만드셨고, 어떻게 표준을 지키고 계시는지 궁금합니다!

**Comment by chang12** [comment]
결국 어떤 google form? 같은 장치가 있으면 좋겠어요.

google form 으로 설문에 답하듯이,
물어보는 것들에 순차적으로 답을 입력하면,
- (회사가 운영하는 product 들 목록을 나열하며) 어떤 product 에 대한 로그 인가요?
- (선택한 product 를 구성하는 화면 목록을 나열하며) 어떤 화면에서 발생하는 로그 인가요?
- (view, click, ... 등 행동 목록을 나열하며) 어떤 행동에 대한 로그 인가요?
- 등등

로그 schema 들이 모이는 어떤 database 에 row 가 insert 되고,
해당 row 에 대한 id? link? 같은게 발급 되고,
로깅 관련 개발을 진행할 engineer 에게 전달하면 되는.
